workspaces/arborist/lib/arborist/reify.js

```diff
@@ -716,8 +716,15 @@ module.exports = cls => class Reifier extends cls {
     // ${REGISTRY} or something.  This has to be threaded through the
     // Shrinkwrap and Node classes carefully, so for now, just treat
     // the default reg as the magical animal that it has been.
-    return resolved && resolved
-      .replace(/^https?:\/\/registry\.npmjs\.org\//, this.registry)
+    const resolvedURL = new URL(resolved)
+    if (resolved
+      && ((this.options.replaceRegistryHost === 'npmjs'
+        && resolvedURL.origin === 'https://registry.npmjs.org')
+      || this.options.replaceRegistryHost === 'always')
+    ) {
+      return new URL(resolvedURL.pathname, this.registry).href
```

Comment:
or HTTP basic auth, or a port, or a protocol?

Reply:
the more i think about it the more i see where this is doing a lot of things differently..

1. this will no longer replace `http://registry.npmjs.org` while the old code did
2. if the replacement registry contains a path, that will be lost here while the old code would keep it
3. if the resolved field has a query parameter, that gets lost

basic auth, port and protocol will end up being populated by whatever `this.registry` has, so i think those are fine as-is

Reply:
I'll address those differences.